aviary/subsystems/propulsion/hamilton_standard.py

```diff
@@ -0,0 +1,932 @@
+
+        self.add_output('thrust_coefficient_comp_loss',
+                        val=np.zeros(nn), units='unitless')
+        self.add_output('Thrust', val=np.zeros(nn), units='lbf')
+        # keep them for reporting but don't seem to be required
+        self.add_output('propeller_efficiency', val=np.zeros(nn), units='unitless')
+        self.add_output('install_efficiency', val=np.zeros(nn), units='unitless')
+
+    def setup_partials(self):
+        arange = np.arange(self.options['num_nodes'])
+
+        self.declare_partials('thrust_coefficient_comp_loss', [
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+        ], rows=arange, cols=arange)
+        self.declare_partials('Thrust', [
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+            Dynamic.Mission.PROPELLER_TIP_SPEED,
+            'density_ratio',
+            Dynamic.Mission.INSTALLATION_LOSS_FACTOR,
+        ], rows=arange, cols=arange)
+        self.declare_partials('Thrust', [
+            Aircraft.Engine.PROPELLER_DIAMETER,
+        ])
+        self.declare_partials('propeller_efficiency', [
+            'adv_ratio',
+            'power_coefficient',
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+        ], rows=arange, cols=arange)
+        self.declare_partials('install_efficiency', [
+            'adv_ratio',
+            'power_coefficient',
+            'thrust_coefficient',
+            'comp_tip_loss_factor',
+            Dynamic.Mission.INSTALLATION_LOSS_FACTOR,
+        ], rows=arange, cols=arange)
+
+    def compute(self, inputs, outputs):
+        ctx = inputs['thrust_coefficient']*inputs['comp_tip_loss_factor']
+        outputs['thrust_coefficient_comp_loss'] = ctx
+        diam_prop = inputs[Aircraft.Engine.PROPELLER_DIAMETER][0]
+        tipspd = inputs[Dynamic.Mission.PROPELLER_TIP_SPEED]
+        install_loss_factor = inputs[Dynamic.Mission.INSTALLATION_LOSS_FACTOR]
+        outputs['Thrust'] = ctx*tipspd**2*diam_prop**2 * \
+            inputs['density_ratio']/(1.515E06)*364.76*(1. - install_loss_factor)
+        outputs['propeller_efficiency'] = inputs['adv_ratio'] * \
+            ctx/inputs['power_coefficient']
+        outputs['install_efficiency'] = outputs['propeller_efficiency'] * \
+            (1. - install_loss_factor)
+
+    def compute_partials(self, inputs, partials):
+        XFT = inputs['comp_tip_loss_factor']
+        ctx = inputs['thrust_coefficient']*XFT
+        diam_prop = inputs[Aircraft.Engine.PROPELLER_DIAMETER][0]
+        install_loss_factor = inputs[Dynamic.Mission.INSTALLATION_LOSS_FACTOR]
+        tipspd = inputs[Dynamic.Mission.PROPELLER_TIP_SPEED]
+
+        partials["thrust_coefficient_comp_loss", 'thrust_coefficient'] = XFT
+        partials["thrust_coefficient_comp_loss",
+                 'comp_tip_loss_factor'] = inputs['thrust_coefficient']
+        partials["Thrust", 'thrust_coefficient'] = XFT*tipspd**2*diam_prop**2 * \
```

Comment:
Probably better for readability to put `1.0 / /(1.515E06) * 364.76` into a temp variable `fact1` or something.

Reply:
Done, good idea!

Reply:
is this a unit conversion factor? it might be better to have a more descriptive name than fact1